index.js

```diff
@@ -1,14 +1,46 @@
 'use strict'
 
 var ChangesStream = require('changes-stream')
+var clean = require('normalize-registry-metadata')
+var through2 = require('through2')
 var hypercore = require('hypercore')
 var swarm = require('hyperdrive-archive-swarm')
 var ndjson = require('ndjson')
 var pump = require('pump')
+var afterAll = require('after-all')
 var levelup = require('level')
+var sub = require('subleveldown')
+
+var normalize = through2.obj(function (change, env, cb) {
+  var stream = this
+  var doc = change.doc
+  var next = afterAll(cb)
+
+  clean(doc)
+
+  if (!doc) {
+    console.log('skipping %s - invalid document', change.id)
+  } else if (!doc.versions || doc.versions.length === 0) {
+    console.log('skipping %s - no versions detected', change.id)
+  } else {
+    Object.keys(doc.versions).forEach(function (version) {
+      var key = doc.name + '@' + version
+      var done = next()
+      index.get(key, function (err, value) {
+        if (!err || !err.notFound) return done(err)
+        stream.push(doc.versions[version])
+        index.put(key, true, function (err) {
```

Comment:
@mafintosh what happens to the hypercore stream if this put command isn't successful, the code is run again and this therefore same version is pushed to the hypercore stream twice?